numpy/lib/index_tricks.py

```diff
@@ -195,12 +196,12 @@ def __getitem__(self, key):
             if start is None:
                 start = 0
             if isinstance(step, (_nx.complexfloating, complex)):
-                step = abs(step)
-                length = int(step)
+                # Prevent the (potential) creation of integer arrays
+                step = length = int(abs(step))
                 if step != 1:
                     step = (key.stop-start)/float(step-1)
-                stop = key.stop + step
-                return _nx.arange(0, length, 1, float)*step + start
+                typ = _nx.result_type(0.0, start, stop, step)
```

Comment:
```suggestion
                typ = _nx.result_type(0, start, stop, step)
```
for consistency with the no-comma case. Almost certainly makes no difference, so all the more reason to be consistent.

Reply:
Wait nevermind, I think the bug is the other way around.

Reply:
This does actually make a difference here. 
As to remain consisting with numpy `<= 1.19` the return dtype _must_ be float when `step` is a complex number. 
The complex `step` is converted into an integer a few lines earlier, so without the `0.0` this could lead to the unwanted situation where `result_type()` returns an integer type.

Reply:
> Wait nevermind, I think the bug is the other way around.

The other way around is still fine btw, as `step` is never converted into an integer when determining the return dtype.

``` python
                if isinstance(step, (_nx.complexfloating, complex)):
                    step = abs(step)
                    size.append(int(step))
```